lib/xcodeproj/scheme/environment_variables.rb

```diff
@@ -0,0 +1,149 @@
+require 'xcodeproj/scheme/xml_element_wrapper'
+
+module Xcodeproj
+  class XCScheme
+    VARIABLES_NODE = 'EnvironmentVariables'
+    VARIABLE_NODE = 'EnvironmentVariable'
+
+    # This class wraps the EnvironmentVariables node of a .xcscheme XML file. This
+    # is just a container of EnvironmentVariable objects. It can either appear on a
+    # LaunchAction or TestAction scheme group.
+    #
+    class EnvironmentVariables < XMLElementWrapper
+      # @param [nil,REXML::Element,Array<EnvironmentVariable>,Array<Hash{Symbol => String,Bool}>] node_or_variables
+      #   The 'EnvironmentVariables' XML node that this object represents. If nil, it will create a default XML
+      #   node to use.
+      # @note If initializing with an array of hashes, the hash should be of the form
+      #   {:key => <String>, :value => <String>(, :enabled => <Bool>)}.
+      # @see EnvironmentVariable#initialize
+      #
+      def initialize(node_or_variables = nil)
+        create_xml_element_with_fallback(node_or_variables, VARIABLES_NODE) do
+          @all_variables = []
+          node_or_variables.each { |var| assign_variable(var) } unless node_or_variables.nil?
+        end
+      end
+
+      # @return [Array<EnvironmentVariable>]
+      #   The key value pairs currently set in @xml_element
+      #
+      def all_variables
+        @all_variables ||= @xml_element.get_elements(VARIABLE_NODE).map { |variable| EnvironmentVariable.new(variable) }
+      end
+
+      # Adds a given variable to the set of environment variables, or replaces it if that key already exists
+      #
+      # @param [EnvironmentVariable,Hash{Symbol => String,Bool}] variable
+      #   The variable to set
+      # @note If initializing with an array of hashes, the hash should be of the form
+      #   {:key => <String>, :value => <String>(, :enabled => <Bool>)}.
+      # @see EnvironmentVariable#initialize
+      # @return [Array<EnvironmentVariable>]
+      #   The new set of environment variables after addition
+      #
+      def assign_variable(variable)
```

Comment:
I don't think so, since it would return an `EnvironmentVariable` anyway, I'd say let it access all variables (enabled or disabled). The user could still check the variable state once retrieved.